jenkins_job_manager/raw_ext.py

```diff
@@ -0,0 +1,81 @@
+"""
+Contains extensions for raw xml loading with jjb
+"""
+import os
+import logging
+import jinja2
+import xml.etree.ElementTree
+import jenkins_jobs.modules.base
+from jenkins_jobs.xml_config import XmlJob, XmlJobGenerator
+
+log = logging.getLogger("jjm")
+
+
+def xml_escape(s):
+    """hacks ElementTree to give us strings escaped for xml"""
+    ET = xml.etree.ElementTree
+    a = ET.Element("a")
+    a.text = s
+    # spit out element as string, slice out the element text
+    escaped_s = ET.tostring(a, encoding="unicode")[3:-4]
+    return escaped_s
+
+
+def load_xml_escaped(path):
+    with open(path) as fp:
+        contents = fp.read()
+    return xml_escape(contents)
+
+
+# initialize a global jinja env
+jenv = jinja2.Environment(
+    loader=jinja2.FileSystemLoader(searchpath=(os.curdir,)),
+    undefined=jinja2.StrictUndefined,
```

Comment:
Using jinja2 templates with autoescape=False is dangerous and can lead to XSS. Use autoescape=True or use the select_autoescape function to mitigate XSS vulnerabilities.

Reply:
yeah, maybe having autoescape disabled was leading to the include bug we saw. I'll give it a shot real quick since that "xml_escape" is a big hack.

Reply:
yep. Jinja has some bug around entity escapes from `include` directives. I noticed in on instance `includes` are handled internally as generators. So the only way to get it to render correctly is:

```
{% set groovy_script %}{% include "some_groovy_script.groovy" %}{% endset -%}
    {{ groovy_script | forceescape }}
```

other methods like `{% filter forceescape %}{% include ...` or using a macro wrapper won't evaluate correctly.